internal/repo/repo.go

```diff
@@ -1,9 +1,106 @@
 package repo
 
-import "github.com/ozoncp/ocp-team-api/internal/models"
+import (
+	"context"
+	sq "github.com/Masterminds/squirrel"
+	"github.com/jmoiron/sqlx"
+	"github.com/ozoncp/ocp-team-api/internal/models"
+)
+
+const (
+	tableName = "team"
+)
 
 type Repo interface {
-	AddTeams(teams []models.Team) error
-	ListTeams(limit, offset uint64) ([]models.Team, error)
-	DescribeTeam(teamId uint64) (*models.Team, error)
+	AddTeam(ctx context.Context, team models.Team) (uint64, error)
+	AddTeams(ctx context.Context, teams []models.Team) error
+	GetTeam(ctx context.Context, teamId uint64) (*models.Team, error)
+	ListTeams(ctx context.Context, limit, offset uint64) ([]models.Team, error)
+	RemoveTeam(ctx context.Context, teamId uint64) error
+}
+
+func NewRepo(db *sqlx.DB) Repo {
+	return &repo{db}
+}
+
+type repo struct {
+	db *sqlx.DB
+}
+
+func (r *repo) AddTeam(ctx context.Context, team models.Team) (uint64, error) {
+	query := sq.Insert(tableName).
+		Columns("name", "description").
+		Values(team.Name, team.Description).
+		Suffix("RETURNING id").
+		RunWith(r.db).
+		PlaceholderFormat(sq.Dollar)
+
+	err := query.QueryRowContext(ctx).Scan(&team.Id)
+
+	return team.Id, err
+}
+
+func (r *repo) AddTeams(ctx context.Context, teams []models.Team) error {
+	query := sq.Insert(tableName).
+		Columns("name", "description").
+		RunWith(r.db).
+		PlaceholderFormat(sq.Dollar)
+
+	for _, team := range teams {
+		query = query.Values(team.Name, team.Description)
+	}
+
+	_, err := query.ExecContext(ctx)
+	return err
+}
+
+func (r *repo) GetTeam(ctx context.Context, teamId uint64) (*models.Team, error) {
+	query := sq.Select("*").
+		From(tableName).
+		Where(sq.Eq{"id": teamId}).
+		RunWith(r.db).
+		PlaceholderFormat(sq.Dollar)
+
+	var team models.Team
+	if err := query.QueryRowContext(ctx).Scan(&team.Id, &team.Name, &team.Description); err != nil {
+		return nil, err
+	}
+
+	return &team, nil
+}
+
+func (r *repo) ListTeams(ctx context.Context, limit, offset uint64) ([]models.Team, error) {
+	query := sq.Select("*").
+		From(tableName).
+		RunWith(r.db).
+		Limit(limit).
+		Offset(offset).
+		PlaceholderFormat(sq.Dollar)
+
+	rows, err := query.QueryContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var teams []models.Team
+	for rows.Next() {
+		var team models.Team
+		if err := rows.Scan(&team.Id, &team.Name, &team.Description); err != nil {
+			continue
```

Comment:
Буду верить, что это опечатка:)
```suggestion
			return nil, err
```

Reply:
она самая :)
поправил